### ISSUE TITLE
mixin: networking overview renders useless panels for the gateway

### ISSUE THREAD

**Issue by QuentinBisson** [grafana/mimir#7911]
#### Describe the bug

The networking overview dashboards renders the number of inflight request for the mimir gateway https://github.com/grafana/mimir/blob/1e903e4620504a265d49ccee621f09e17b3a18f8/operations/mimir-mixin/dashboards/overview-networking.libsonnet#L9 but the gateway(when using the non enterprise gateway which is nginx) will never expose the metrics used in those panel:
- Inflight requests (per pod) uses `cortex_inflight_requests`
- TCP connections (per pod) uses `cortex_tcp_connections` and `cortex_tcp_connections_limit`

Should those panels be removed for the gateway in the case that it uses nginx?

On that note, the rollout dashboard should probably be removable from the mixins as it's main use case it for the enterprise gateway

#### To Reproduce

1. Start Mimir (SHA or version)
2. Perform Operations(Read/Write/Others)

#### Environment

- Infrastructure: [e.g., Kubernetes, bare-metal, laptop]
- Deployment tool: [e.g., helm, jsonnet]

**Comment by dimitarvdimitrov** [comment]
yeah i think it makes sense to remove those panels by default in the rendered dashboards. They are removed via `$._config.gateway_enabled`. I believe this is the config that's used for the config default dashboards https://github.com/grafana/mimir/blob/9090af52a8e104c37bb4da975bbefbf5185458cc/operations/mimir-mixin/mixin-compiled.libsonnet

**Comment by parambath92** [comment]
Hi @dimitarvdimitrov , Can I work on this ?
